packages/search/src/search.tsx

```diff
@@ -90,349 +96,349 @@ type State = {
-		if ( ! inputElement.createTextRange ) {
-			return inputElement.scrollLeft;
-		}
+			setState( {
+				keyword: '',
+				isOpen: false,
+			} );
 
-		const range = inputElement.createTextRange();
-		const inputStyle = window.getComputedStyle( inputElement, undefined );
-		const paddingLeft = parseFloat( inputStyle.paddingLeft );
-		const rangeRect = range.getBoundingClientRect();
-		const scrollLeft =
-			inputElement.getBoundingClientRect().left +
-			inputElement.clientLeft +
-			paddingLeft -
-			rangeRect.left;
-		return scrollLeft;
-	};
+			if ( searchInput.current ) {
+				searchInput.current.value = ''; // will not trigger onChange
+			}
 
-	focus = (): void => {
-		// if we call focus before the element has been entirely synced up with the DOM, we stand a decent chance of
-		// causing the browser to scroll somewhere odd. Instead, defer the focus until a future turn of the event loop.
-		setTimeout( () => this.searchInput.current?.focus(), 0 );
-	};
+			if ( pinned ) {
+				searchInput.current?.blur();
+				openIcon.current?.focus();
+			} else {
+				searchInput.current?.focus();
+			}
 
-	blur = (): void => this.searchInput.current?.blur();
+			onSearchClose?.( event );
 
-	clear = (): void => this.setState( { keyword: '' } );
+			recordEvent?.( 'Clicked Close Search' );
+		},
+		[ disabled, pinned, onSearchClose, recordEvent ]
+	);
 
-	onBlur = ( event: FocusEvent< HTMLInputElement > ): void => {
-		if ( this.props.onBlur ) {
-			this.props.onBlur( event );
-		}
+	const closeListener = React.useMemo( () => keyListener( closeSearch ), [ closeSearch ] );
+	const openListener = React.useMemo( () => keyListener( openSearch ), [ openSearch ] );
 
-		this.setState( { hasFocus: false } );
-	};
+	const scrollOverlay = React.useCallback( (): void => {
+		overlay &&
+			window.requestAnimationFrame( () => {
+				if ( overlay.current && searchInput.current ) {
+					overlay.current.scrollLeft = getScrollLeft( searchInput.current );
+				}
+			} );
+	}, [] );
 
-	onChange = ( event: ChangeEvent< HTMLInputElement > ): void => {
-		this.setState( {
-			keyword: event.target?.value ?? this.state.keyword,
-		} );
-	};
+	React.useEffect( () => {
+		scrollOverlay();
+	}, [ keyword, isOpen, hasFocus, scrollOverlay ] );
 
-	keyUp = ( event: KeyboardEvent< HTMLInputElement > ): void => {
-		if ( event.key === 'Enter' && window.innerWidth < 480 ) {
-			//dismiss soft keyboards
-			this.blur();
-		}
+	const onBlur = React.useCallback(
+		( event: FocusEvent< HTMLInputElement > ): void => {
+			if ( onBlurProp ) {
+				onBlurProp( event );
+			}
 
-		if ( ! this.props.pinned ) {
-			return;
-		}
+			setState( { hasFocus: false } );
+		},
+		[ onBlurProp ]
+	);
 
-		if ( event.key === 'Escape' ) {
-			this.closeListener( event );
-		}
-		this.scrollOverlay();
-	};
+	const onChange = React.useCallback(
+		( event: ChangeEvent< HTMLInputElement > ): void => {
+			setState( {
+				keyword: event.target?.value ?? keyword,
```

Comment:
Can it really happen that `event.target.value` is null-ish and the `keyword` fallback is used? The `onChange` handler is attached as `<input type="search" onChange />`. `event.target` is always there (even the type of `ChangeEvent` agrees with that) and the value is always a string.

Reply:
I thought that `target` could be null 🤔 But you're right that it can't according to the type so 🤷🏼‍♀️